pkg/blockbuilder/blockbuilder.go

```diff
@@ -35,6 +35,8 @@ import (
 	"github.com/grafana/mimir/pkg/util/validation"
 )
 
+var tracer = otel.Tracer("pkg/blockbuilder")
```

Comment:
Is it guaranteed that this initialization runs after Otel gets parameters from config?
Would it be worth having some kind of `sync.Once` thing called when it is first needed?

Reply:
I was wondering same question, because this definitely runs at package initialization time and we configure tracer for the first time from `main()`.

So I dug the code, and found out that the default implementation of the tracer is a special one which is `sync.Once`-replaced to the first one you manually configure. 

That's actually an inconvenience for tests, but apparently having per-package global tracers is the way to go so I won't fight it.